odoo/addons/test_new_api/models/test_new_api.py

```diff
@@ -833,3 +833,107 @@ class City(models.Model):
 
     name = fields.Char()
     country_id = fields.Many2one('test_new_api.country')
+
+
+class SelectionBaseRequired(models.Model):
+    _name = 'test_new_api.model_selection_base_required'
+    _description = "Dummy model with a base selection field"
+
+    active = fields.Boolean(default=True)
+    my_selection = fields.Selection([
+        ('foo', "Foo"),
+        ('bar', "Bar"),
+    ], required=True, default=lambda r: 'foo')
+
+
+class SelectionExtensionDefault(models.Model):
+    _inherit = 'test_new_api.model_selection_base_required'
+    _description = "Dummy model with a selection field extension with ondelete default"
+
+    my_selection = fields.Selection(selection_add=[
+        ('baz', "Baz"),
+    ], ondelete={'baz': 'default'})
+
+
+class SelectionExtensionCascade(models.Model):
+    _inherit = 'test_new_api.model_selection_base_required'
+    _description = "Dummy model with a selection field extension with ondelete cascade"
+
+    my_selection = fields.Selection(selection_add=[
+        ('eggs', "Eggs"),
+    ], ondelete={'eggs': 'cascade'})
+
+
+class SelectionExtensionLiteral(models.Model):
+    _inherit = 'test_new_api.model_selection_base_required'
+    _description = "Dummy model with a selection field extension with ondelete set <option>"
+
+    my_selection = fields.Selection(selection_add=[
+        ('bacon', "Bacon"),
+    ], ondelete={'bacon': 'set bar'})
+
+
+class SelectionExtensionMultipleExplicit(models.Model):
+    _inherit = 'test_new_api.model_selection_base_required'
+    _description = "Dummy model with a selection field extension with ondelete set <option>"
+
+    my_selection = fields.Selection(selection_add=[
+        ('pikachu', "Pikachu"),
+        ('eevee', "Eevee"),
+    ], ondelete={'pikachu': 'default', 'eevee': 'set bar'})
+
+
+class SelectionExtensionMultipleImplicit(models.Model):
+    _inherit = 'test_new_api.model_selection_base_required'
+    _description = "Dummy model with a selection field extension with ondelete set <option>"
+
+    my_selection = fields.Selection(selection_add=[
+        ('charmander', "Charmander"),
+        ('squirtle', "Squirtle"),
+        ('bulbasaur', "Bulbasaur"),
+    ], ondelete={'charmander': 'default', 'squirtle': 'default', 'bulbasaur': 'default'})
```

Comment:
I don't see the difference between this extension and the former one.
Can you explain?

Reply:
I think this is a leftover of the first implementation where you could simply have ondelete be a string or a list of strings, this one would test the list of strings